### ISSUE TITLE
Clean install from conda does not include pystac_client

### ISSUE THREAD

**Issue by caitlinadams** [opendatacube/odc-stac#49]
When installing odc-stac into a new conda environment, it does not automatically install [pystac-client](https://github.com/stac-utils/pystac-client). Is it possible to include pystac-client, seeing as each example uses it? 

Conda env commands to reproduce environment:
```
conda create -n odcstac-test python=3.8
conda activate odcstac-test
conda install -c conda-forge odc-stac
```

After attempting to run the example and getting a module import error for pystac-client, I also needed to run
```
pip install pystac-client
```

As a side note, when using the environment created above, the Digital Earth Africa example fails at the plotting step because matplotlib isn't in the environment. I think this is probably a separate issue, but seeing as the example uses xarray's plotting functionality, I was wondering why it's possible to work with xarrays as part of loading using odc-stac, but not possible to leverage xarray's plotting methods without explicitly installing matplotlib. If you could shed any light on this, I'd be interested!

Relevant traceback:
```
[File ~/opt/anaconda3/envs/testenv/lib/python3.8/site-packages/xarray/plot/utils.py:34, in import_matplotlib_pyplot()
     ]()[32](file:///Users/caitlinadams/opt/anaconda3/envs/testenv/lib/python3.8/site-packages/xarray/plot/utils.py?line=31)[ """import pyplot"""
     ]()[33](file:///Users/caitlinadams/opt/anaconda3/envs/testenv/lib/python3.8/site-packages/xarray/plot/utils.py?line=32)[ # TODO: This function doesn't do anything (after #6109), remove it?
---> ]()[34](file:///Users/caitlinadams/opt/anaconda3/envs/testenv/lib/python3.8/site-packages/xarray/plot/utils.py?line=33)[ import matplotlib.pyplot as plt
     ]()[36](file:///Users/caitlinadams/opt/anaconda3/envs/testenv/lib/python3.8/site-packages/xarray/plot/utils.py?line=35)[ return plt

ModuleNotFoundError: No module named 'matplotlib']()
```

**Comment by Kirill888** [comment]
@caitlinadams thanks for the report, but `odc-stac` does not use `pystac-client`, it's a common source of stac items for it, but there is no direct use of it within the library and so it's not included as dependency in pip or conda.

Examples often assume presence of other libraries, like `matplotlib` listed in your trace. Sample python environment is available here:

- https://github.com/opendatacube/odc-stac/blob/develop/binder/environment.yml
- https://github.com/opendatacube/odc-stac/blob/develop/tests/test-env-py38.yml

but they are not minimal or anything. I guess we can add a section to docs listing libraries needed to run examples.

**Comment by caitlinadams** [comment]
Thanks for letting me know, @Kirill888 -- much appreciated. I will look at producing a minimal environment.yml file needed to run all of the examples, as it should help people get off and running. I'll send a pull request when I've done that.
